articles/mvc-mvp-mvvm/index.md

```diff
@@ -0,0 +1,52 @@
+
+Here the developers can work on different parts of the pattern separately. Hence, work can be completed a lot quicker. This pattern can come in handy if multiple views are required for the application. Many of the elements function together and coordinate with other components. However, since they all rely on each other, if one of the components fails, it may pose a problem for the whole application.
+
+#### Model View Presenter
+
+Here the controller part is replaced by the presenter. The **Presenter** is responsible for accepting all the incoming inputs/requests from the user. It then transfers them to the **Model**. The model accesses the data the user wants and sends it back as a result to the **View**. Hence, the key task of the presenter is to deal with all the UI events so that users can retrieve their data. The role of model and view in [MVP](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter) is the same as in MVC. One of the assisting factors to the application while using MVP is that the view is submissive. Its sole purpose is to approve all user activities and transfer the rest of the work to the presenter. Additionally, since all three parts are separated, debugging becomes simpler because unit testing can be done when other parts are still being developed.
+
+#### Model View View-Model
+
+In [MVVM](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel), the view is independent of the model. This means that the UI part is separated from the core logic of the application. It's the job of the view model to retrieve the data from the model and present them in the view. Also, the view model carries out the computational work of the view by handling the user events and updating the UI. This is because in MVVM the view is responsible for notifying the view-model about all the user action. As a result, we can assume that View and View-Model has a two-way data binding relationship.
+
+### Mind Map
+
+Since the above architectural designs are based around the idea of collaborating with their own structure to achieve a positive outcome, each component has its own set of responsibilities. Model and View are shared by all of the patterns mentioned above. The third aspect is what distinguishes them from one another. As a result, here is a simplified mind map to help you visualize what each pattern represents.
+
+![Architectural Patterns](/engineering-education/mvc-mvp-mvvm/img.jpg)

```

Comment:
Include image source attributions and a title for the image

Reply:
> Include image source attributions and a title for the image

@cyberShaw That image was not taken from the internet. I've designed it myself using diagrams.net